clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/cache/CatsOnDemandCacheUpdater.groovy

```diff
@@ -93,6 +96,8 @@ class CatsOnDemandCacheUpdater implements OnDemandCacheUpdater {
         log.warn("$agent.providerName/$agent.onDemandAgentType failed to handle on demand update for $type", e)
       }
     }
+
+    return hasOnDemandResults ? OnDemandCacheUpdater.OnDemandCacheStatus.PENDING : OnDemandCacheUpdater.OnDemandCacheStatus.SUCCESSFUL
```

Comment:
I'm confused about the semantics here. Any result returned from `agent.handle` is written to the cache [here](https://github.com/spinnaker/clouddriver/pull/284/files#diff-52b84ddde4b53349570a957e33dce33fL77). Why does that qualify the result as `PENDING`?